src/jit/inline.cpp

```diff
@@ -0,0 +1,93 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "jitpch.h"
+#ifdef _MSC_VER
+#pragma hdrstop
+#endif
+
+static const char* InlineExplanations[] = 
+{
+    "unused initial",
+#define INLINE_OBSERVATION(name, type, description, impact, target) description,
+#include "inline.def"
+#undef INLINE_OBSERVATION
+    "unused final"
+};
+
+static const InlineTarget InlineTargets[] =
+{
+    InlineTarget::CALLSITE,
+#define INLINE_OBSERVATION(name, type, description, impact, target) InlineTarget::target,
+#include "inline.def"
+#undef INLINE_OBSERVATION
+    InlineTarget::CALLSITE
+};
+
+static const InlineImpact InlineImpacts[] =
+{
+    InlineImpact::FATAL,
+#define INLINE_OBSERVATION(name, type, description, impact, target) InlineImpact::impact,
+#include "inline.def"
+#undef INLINE_OBSERVATION
+    InlineImpact::FATAL,
+};
+
+const char* GetDescriptionString(InlineObservation obs)
+{
+    assert((obs > InlineObservation::UNUSED_INITIAL) &&
+           (obs < InlineObservation::UNUSED_FINAL));
```

Comment:
Should there be an IsLegalInlineObservation(InlineObservation obs) function with this expression? Then you could have: assert(IsLegalInlineObservation(obs)) instead of repeating this expression